deployment/src/main/java/com/datastax/oss/quarkus/deployment/internal/CassandraClientProcessor.java

```diff
@@ -206,26 +197,30 @@ void setupSslSupport(
   List<ReflectiveClassBuildItem> registerMetricsFactoriesForReflection(
       CassandraClientBuildTimeConfig buildTimeConfig,
       Optional<MetricsCapabilityBuildItem> metricsCapability) {
+
     if (buildTimeConfig.metricsEnabled && metricsCapability.isPresent()) {
+
+      Stream<String> clzStream = Stream.empty();
       MetricsCapabilityBuildItem metricsCapabilityItem = metricsCapability.get();
       if (metricsCapabilityItem.metricsSupported(MetricsFactory.MICROMETER)) {
-        return Arrays.asList(
-            new ReflectiveClassBuildItem(
-                true,
-                true,
-                "com.datastax.oss.driver.internal.metrics.micrometer.MicrometerMetricsFactory"),
-            new ReflectiveClassBuildItem(true, true, TaggingMetricIdGenerator.class));
+        clzStream =
+            Stream.of(
+                TaggingMetricIdGenerator.class.getName(),
+                "com.datastax.oss.driver.internal.metrics.micrometer.MicrometerMetricsFactory");
       } else if (metricsCapabilityItem.metricsSupported(MetricsFactory.MP_METRICS)) {
-        return Arrays.asList(
-            new ReflectiveClassBuildItem(
-                true,
-                true,
-                "com.datastax.oss.driver.internal.metrics.microprofile.MicroProfileMetricsFactory"),
-            new ReflectiveClassBuildItem(true, true, TaggingMetricIdGenerator.class));
+        clzStream =
+            Stream.of(
+                TaggingMetricIdGenerator.class.getName(),
+                "com.datastax.oss.driver.internal.metrics.microprofile.MicroProfileMetricsFactory");
```

Comment:
It sure _seems_ like we need an additional else case here to handle the situation in which neither if block above is invoked.  That would correlate to metrics of some kind being in play but it being neither Micrometer or MicroProfile.  Is there a guard somewhere else that I missed which prevents this case from coming up?

Reply:
Looking at the javadocs of `metricsSupported`:

```
Returns:
true if this factory supports the named metrics system. Arbitrary strings are allowed. Constants are present for a few.
See Also:
io.quarkus.runtime.metrics.MetricsFactory.MICROMETER, io.quarkus.runtime.metrics.MetricsFactory.MP_METRICS
```

So while it seems that Micrometer and MicroProfile are the only ones officially supported, other metrics providers are in theory accepted.

We should indeed add an `else` block and maybe log a warning that driver metrics won't work for any other provider than the two already handled.

Reply:
Done.  It seemed simplest to just log a message indicating which metric impls were supported.  I went back and forth on the level but this seemed like something worthy of a warn.